web/ui/README.md

```diff
@@ -1,81 +1,38 @@
-in `.promu.yml`, and then `make build` (or build Prometheus using
-`go build ./cmd/prometheus`).
+* `mantine-ui`: The new (3.x) React-based web UI for Prometheus.
+* `react-app`: The old (2.x) React-based web UI for Prometheus.
+* `modules`: Shared npm modules for PromQL code editing via [CodeMirror](https://codemirror.net/), which are used by both React apps and external consumers (like [Thanos](https://thanos.io/)).
+* `static`: The build output directory for both React apps. The files in this directory are compiled into the Prometheus binary unless built-in assets are disabled (see the section on this below).
 
-This will serve all files from your local filesystem. This is for development purposes only.
+The directory also contains helper files for building and compiling the UI assets for both React application versions into the Prometheus binary.
 
-### Using Prebuilt UI Assets
+Prometheus serves the new UI by default, but you can still use the Prometheus server feature flag `--enable-feature=old-ui` to switch back to the old UI for the time being.
 
-If you are only working on the go backend, for faster builds, you can use
-prebuilt web UI assets available with each Prometheus release
-(`prometheus-web-ui-<version>.tar.gz`). This allows you to skip building the UI
-from source.
+While both the `mantine-ui` and `modules` directories are part of the same shared npm workspace, the old UI in the `react-app` directory has been separated out of the workspace setup, since its dependencies were too incompatible to integrate.
 
-1. Download and extract the prebuilt UI tarball:
-   ```bash
-   tar -xvf prometheus-web-ui-<version>.tar.gz -C web/ui
-   ```
-
-2. Build Prometheus using the prebuilt assets by passing the following parameter
-   to `make`:
-   ```bash
-   make PREBUILT_ASSETS_STATIC_DIR=web/ui/static build
-   ```
-
-This will include the prebuilt UI files directly in the Prometheus binary,
-avoiding the need to install npm or rebuild the frontend from source.
-
-## React-app
-
-### Introduction
-
-This directory contains two generations of Prometheus' React-based web UI:
+### Pre-requisites
 
-* `react-app`: The old 2.x web UI
-* `mantine-ui`: The new 3.x web UI
-
-Both UIs are built and compiled into Prometheus. The new UI is served by default, but a feature flag
-(`--enable-feature=old-ui`) can be used to switch back to serving the old UI.
-
-Then you have different npm packages located in the folder `modules`. These packages are supposed to be used by the
-two React apps and also by others consumers (like Thanos).
-
-While most of these applications / modules are part of the same npm workspace, the old UI in the `react-app` directory
-has been separated out of the workspace setup, since its dependencies were too incompatible.
-
-### Pre-requisite
-
-To be able to build either of the React applications, you need:
+To be able to build either of the React applications, you will need:
 
 * npm >= v7
 * node >= v20
```

Comment:
```suggestion
* npm >= v10
* node >= v22
```

Based on recent bugs Bartek and Ben raised concerning prombench, this pre-requisite should be updated too.

Reply:
Updated!